tests/sqlparser_bigquery.rs

```diff
@@ -792,6 +792,31 @@ fn parse_typed_struct_syntax_bigquery() {
         },
         expr_from_projection(&select.projection[1])
     );
+
+    // Keywords may be used as field names.
+    let sql = r#"SELECT STRUCT<key INT64, value INT64, join INT64>(42)"#;
```

Comment:
It doesn't look like BQ will actually accept any keyword as a field name here. Running this query I get

```
Syntax error: Unexpected keyword JOIN at [2:39]
```

I think it's fine for us to succeed in parsing this anyway but it'd be nice for the test to use a query that works in BQ.

On that note, renaming `join` also complains about the missing fields in the constructor:

```
STRUCT type has 3 fields but constructor call has 1 fields at [2:8]
```

Reply:
I updated the test to be less confusing/misleading now. Yeah join specifically doesn't seem to be allowed by bigquery, the keyword mainly refers to the parser's list of keywords vs bigquery's keywords (in the example fields `key` and `value` incorrectly fail to parse otherwise).